Mage.Sets/src/mage/cards/j/JasonBrightGlowingProphet.java

```diff
@@ -0,0 +1,88 @@
+import mage.abilities.effects.common.counter.AddCountersTargetEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+
+/**
+ * @author Cguy7777
+ */
+public final class JasonBrightGlowingProphet extends CardImpl {
+
+    private static final FilterControlledPermanent filter = new FilterControlledPermanent();
+
+    static {
+        filter.add(Predicates.or(SubType.ZOMBIE.getPredicate(), SubType.MUTANT.getPredicate()));
+    }
+
+    public JasonBrightGlowingProphet(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ZOMBIE);
+        this.subtype.add(SubType.MUTANT);
+        this.subtype.add(SubType.ADVISOR);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(3);
+
+        // Whenever a Zombie or Mutant you control dies, if its power was different from its base power, draw a card.
+        this.addAbility(new ConditionalInterveningIfTriggeredAbility(
+                new DiesCreatureTriggeredAbility(new DrawCardSourceControllerEffect(1), false, filter, true),
+                JasonBrightGlowingProphetCondition.instance,
+                "Whenever a Zombie or Mutant you control dies, if its power was different from its base power, draw a card."));
+
+        // Come Fly With Me -- {2}, Sacrifice a creature: Put a +1/+1 counter on target creature you control. It gains flying until end of turn.
+        Ability ability = new SimpleActivatedAbility(
+                new AddCountersTargetEffect(CounterType.P1P1.createInstance()), new GenericManaCost(2));
+        ability.addCost(new SacrificeTargetCost(StaticFilters.FILTER_PERMANENT_CREATURE));
+        ability.addEffect(new GainAbilityTargetEffect(
+                FlyingAbility.getInstance(), Duration.EndOfTurn, "it gains flying until end of turn"));
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        this.addAbility(ability.withFlavorWord("Come Fly With Me"));
+    }
+
+    private JasonBrightGlowingProphet(final JasonBrightGlowingProphet card) {
+        super(card);
+    }
+
+    @Override
+    public JasonBrightGlowingProphet copy() {
+        return new JasonBrightGlowingProphet(this);
+    }
+}
+
+enum JasonBrightGlowingProphetCondition implements Condition {
+    instance;
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = source.getEffects().get(0).getTargetPointer().getFirstTargetPermanentOrLKI(game, source);
```

Comment:
this seems nonstandard but I can't think of any problem with it for custom class

I think typically what we'd do is to just make it part of the trigger condition (as a predicate in the filter) and override the trigger phrase, since the intervening if condition can't change between trigger and resolution

Reply:
> Whenever a Zombie or Mutant you control dies, if its power was different from its base power

Looks like it’s a wrong code. Ability must trigger on any dies events, but custom conditional called for own source only:
![IMG_0404](https://github.com/magefree/mage/assets/8344157/0a6f8b8e-a197-46d9-9fc5-4795c93ff886)

I recommends to write custom predicate instead - it will have access to died permanent.

Reply:
Hmm, or maybe it’s fine due DiesCreatureTriggeredAbility and setTargetPointer. It’s weird usage but workable.

Just make sure it works for other effects/dies, not from card itself.

Reply:
I looked into it, and I think everything was fine, but using a custom predicate seems to be cleaner, so I've decided to do that instead.